### DISCUSSION TITLE
Local build for io.opentelemetry.instrumentation

### DISCUSSION THREAD

**Discussion by Dimagreg** [open-telemetry/opentelemetry-java-instrumentation#12408] [Q&A] [answered]
A recent issue with a PR has been merged to main: #12394 and I would like to use it in my code before a new release will be published. How could I manually build the project and use the io.opentelemetry.instrumentation dependency?

**Comment by steverao** [comment]
> How could I manually build the project and use the io.opentelemetry.instrumentation dependency?

Sure, you can build the project directly, for detail, you can refer to [related document](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/CONTRIBUTING.md#building).

**Reply by Dimagreg** [reply]
Hi, Thanks for the reply! I am still facing issues with installing the dependency for my maven project. I use following command to install the local jar: 
```
mvn install:install-file -Dfile="opentelemetry-javaagent-2.9.0-SNAPSHOT.jar" -DgroupId="io.opentelemetry.instrumentation" -DartifactId="opentelemetry-apache-httpclient-5.2" -Dversion="2.9.0-SNAPSHOT" -Dpackaging=jar
```
But when I try to use the dependency in project i get following error: 
```
The import io.opentelemetry.instrumentation.apachehttpclient cannot be resolved
```
Is there a different command for maven to install the local jar?

**Reply by laurit** [reply]
For maven questions you should consult their documentation or ask from the maven community or stack overflow. Instead of using maven you could also use `./gradlew publishToMavenLocal`

**Reply by Dimagreg** [reply, marked as answer]
Found a solution by adding individual jar files from https://oss.sonatype.org/content/repositories/snapshots/io/opentelemetry/instrumentation/
